var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -289,10 +283,31 @@ def install(self, spec, prefix):
                         os.symlink(src_path, dest_path)
 
     @run_after("install")
-    def link_pkg_config(self):
+    def generate_pkg_config(self):
         if self.spec.satisfies("+pcfile"):
+            mkdirp(self.prefix.lib.pkgconfig)
             versioned_pc_file_name = "lua{0}.pc".format(self.version.up_to(2))
-            symlink(
-                join_path(self.prefix.lib, "pkgconfig", versioned_pc_file_name),
-                join_path(self.prefix.lib, "pkgconfig", "lua.pc"),
+            versioned_pc_file_path = join_path(
+                self.prefix.lib, "pkgconfig", versioned_pc_file_name
             )
+            with open(versioned_pc_file_path, "w") as pcfile:
+                # http://lua.2524044.n2.nabble.com/attachment/7666421/0/pkg-config.patch
+                pcfile.write(
+                    """prefix={0}
```

Comment:
Suggestions to make the template a hidden global in this file. Not sure if black or mypy will allow it, but it would not mess with the indentation as much.

Reply:
I tried to set the template as global variable at the beginning of the file and it passed the `spack style`